shell/platform/android/io/flutter/embedding/engine/FlutterEngineGroupCache.java

```diff
@@ -0,0 +1,98 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Static singleton cache that holds {@link io.flutter.embedding.engine.FlutterEngineGroup}
+ * instances identified by {@code String}s.
+ *
+ * <p>The ID of a given {@link io.flutter.embedding.engine.FlutterEngineGroup} can be whatever
+ * {@code String} is desired.
+ *
+ * <p>{@code FlutterEngineGroupCache} is useful for storing pre-warmed {@link
```

Comment:
this sentence makes sense for FlutterEngine, less for FlutterEngineGroups. e.g. a FlutterEngineGroup can't exactly be "pre-warmed"

Reply:
yes, you are right. I will update